app/controllers/account_activations_controller.rb

```diff
@@ -0,0 +1,14 @@
+class AccountActivationsController < ApplicationController
+  def edit
+    user = User.find_by(email: params[:email])
+    if user && !user.activated? && user.authenticated?(:activation, params[:id])
```

Comment:
1.  user && !user.activated?  sử dụng The Safe Navigation Operator (&.) trong Ruby
2. Tai sao chỗ này em truyền vào  params[:id] em nhỉ

Reply:
Edit user tương ứng với id truyền vào